### PR TITLE
NetBSD fixes. Includes fixes for gcc 4.8

### PR DESCRIPTION
**Pull request by skrll**
github newbie so I'm unsure if I should be requesting pull requests for subsets of commits.

### CODE REVIEW

**Comment by brd** [comment]
Any thoughts on getting this in? It has been open awhile without any comment..

**Comment by popcornmix** [comment]
The PR is a bit of a monster. It would be better if it were split into some smaller PRs.
Perhaps separate:
* cosmetics
* warning removal
* error handling
* functional changes, like the stack size increase, which should include an explanation for why it is necessary
* NetBSD specific makefile changes

Then it may be easier to gradually merge them in.

**Comment by brd** [comment]
Sounds fair. Thank you for you reply.

I am looking at doing similar work for FreeBSD, so I will try and break it down.

**Comment by Ruffio** [comment]
@brd will you break it down?

File: CMakeLists.txt [hunk 1]
```diff
@@ -83,7 +83,10 @@ add_subdirectory(middleware/openmaxil)
 
 # FIXME: this directory needs a more sensible name and better
 # platform factoring
-add_subdirectory(interface/usbdk)
+
+if(NOT CMAKE_SYSTEM_NAME MATCHES "NetBSD")
```
Comment (luked99) [inline CMakeLists.txt]: I think this directory has been removed now entirely.

**Comment by Ruffio** [comment]
Is this PR still relevant?

**Comment by skrll** [comment]
If anything is missing I'll do it differently (i.e. better)

**Pull request closed without merging**